src/workloads/scale/MultiPlanStormSortSkip.yml

```diff
@@ -0,0 +1,123 @@
+  large number of documents.
+
+Keywords:
+- memory stress
+- multi-planning
+- sort
+- skip
+- oom
+- out of memory
+
+GlobalDefaults:
+  MaxPhases: &MaxPhases 2
+  # The amount of documents here is important, since trial-runs in the query planner "tournament"
+  # will only use ~30% of total document count increasing the number of total documents helps
+  # reproduce the issue.
+  TotalDocs: &TotalDocs 10000000
+  # To avoid completing the sort-skip early, the number of skipped documents must be large enough,
+  # and should approach the number of documents matching the filtering predicate.
+  # In the following, values similar to those seen during the OOM scenario from HELP-43886 are chosen.
+  MatchingDocs: &MatchingDocs 20000
+  SkippedDocs: &SkippedDocs 18000
+  # Number of random documents
+  RandomDocs: &RandomDocs
+    ^NumExpr:
+      withExpression: "num_total_docs - num_matching_docs"
+      andValues: {num_total_docs: *TotalDocs, num_matching_docs: *MatchingDocs}
+  # The size of the document media field is tuned so that MatchingDocs * DocSize approaches 100MB,
+  # which is the maximum amount of memory that can be used by a sort before spilling to disk.
+  DocMediaSize: &DocMediaSize 4800 # Bytes
+
+Clients:
+  Default:
+    QueryOptions:
+      maxPoolSize: 1100
+      socketTimeoutMS: 7_200_000  # = 2 hour
+      connectTimeoutMS: 7_200_000
+
+Actors:
+- Name: InsertData
+  Type: Loader
+  Threads: 100
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Database: &DB test
+        Blocking: None
+        MultipleThreadsPerCollection: true
+        CollectionCount: 1
+        DocumentCount: *TotalDocs
+        BatchSize: 1000
+        Document:
+          uniqueId: {^Inc: {start: 1}}
+          media: {^FastRandomString: {length: *DocMediaSize}}
+          hasDeducteeChanged: true
+          hasPanChanged: false
+          codes: {
+            ^Choose: {
+              from: [
+                {
+                  tdsReturnId: "ffffffffffffffffffffffff",
+                  tanId: "abcdefgh"
+                },
+                {
+                  tdsReturnId: {^FastRandomString: {length: 24}},
+                  tanId: {^FastRandomString: {length: 8}}
+                }
+              ],
+              weights: [*MatchingDocs, *RandomDocs]
+            }
+          }
+        Indexes:
+          # 7 indexes will be considered during multi-planning.
+          # 5 indexes involve the sorter, consuming up to 500MB per query.
+          - keys: {codes.tdsReturnId: 1, _id: -1}
+          - keys: {codes.tanId: 1, _id: -1}
+          - keys: {uniqueId: 1}
+          - keys: {codes.tdsReturnId: 1, hasDeducteeChanged: 1}
+          - keys: {codes.tanId: 1, codes.tdsReturnId: 1}
+          - keys: {codes.tanId: 1, hasPanChanged: 1}
+          - keys: {codes.tdsReturnId: 1, codes.uniqueId: 1}
+          - keys: {codes.tanId: 1, uniqueId: 1}
+
+- Name: MultiPlanStorm
```

Comment:
How long does this phase roughly take to OOM the system? What about the whole workload, including the insertion phase?

Reply:
It takes ~10min to insert the data and build the indexes. After the InsertData phase completes, around 10s are needed to OOM the server in the MultiPlanStorm phase.

Reply:
Sounds good, thanks!